front/pages/api/w/[wId]/vaults/[vId]/data_sources/[dsId]/index.ts

```diff
@@ -0,0 +1,72 @@
+import type {
+  DataSourceOrViewInfo,
+  WithAPIErrorResponse,
+} from "@dust-tt/types";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { getDataSourceInfo } from "@app/lib/api/vaults";
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import { apiError } from "@app/logger/withlogging";
+
+export type GetDataSourceResponseBody = {
+  dataSource: DataSourceOrViewInfo;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<GetDataSourceResponseBody>>,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  const dataSource = await DataSourceResource.fetchByName(
+    auth,
+    req.query.dsId as string
+  );
+
+  const vault = dataSource?.vault;
+
+  if (
+    !dataSource ||
+    !vault ||
+    req.query.vId !== vault.sId ||
+    (!auth.isAdmin() && !auth.hasPermission([vault.acl()], "read"))
+  ) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "data_source_not_found",
+        message: "The data source you requested was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET": {
+      return res.status(200).json({
+        dataSource: getDataSourceInfo(dataSource),
```

Comment:
This should definitely be on the resource IMHO.

Reply:
as discussed irl category is specific to the endpoint (and so "info), should not be moved in the resource